src/credoProvider.ts

```diff
@@ -109,34 +115,62 @@ export default class ElixirLintingProvider {
             return;
         }
 
-        let decoded = "";
-        const diagnostics: any[] = [];
+        this.getLintedFiles(vscode, (files) => {
+          if (files.indexOf(textDocument.fileName) === -1) {
+            return;
+          }
+
+          let decoded = "";
+          const diagnostics: any[] = [];
+
+          let args =  ["credo", "list", "--format=oneline", "--read-from-stdin"];
+
+          const settings = vscode.workspace.getConfiguration("elixirLinter");
+          if (settings.useStrict === true) {
+              args = args.concat("--strict");
+          }
+
+          // use stdin for credo to prevent running on entire project
+          const childProcess = cp.spawn(ElixirLintingProvider.linterCommand, args, cmd.getOptions(vscode));
+          childProcess.stdin.write(textDocument.getText());
+          childProcess.stdin.end();
+
+          if (childProcess.pid) {
+              childProcess.stdout.on("data", (data: Buffer) => {
+                  decoded += data;
+              });
+              childProcess.stdout.on("end", () => {
+                  this.parseOutput(decoded).forEach( (item) => {
+                      if (item) {
+                          diagnostics.push(this.getDiagnosis(item, vscode));
+                      }
+
+                  });
+                  this.diagnosticCollection.set(textDocument.uri, diagnostics);
+              });
+          }
+        });
 
-        let args =  ["credo", "list", "--format=oneline", "--read-from-stdin"];
+    }
 
-        const settings = vscode.workspace.getConfiguration("elixirLinter");
-        if (settings.useStrict === true) {
-            args = args.concat("--strict");
-        }
+    private getLintedFiles(vscode = this.vscode, callback: (files: string[]) => void) {
+        let decoded = "";
+        const args =  ["credo", "info", "--verbose", "--format=json"];
 
-        // use stdin for credo to prevent running on entire project
         const childProcess = cp.spawn(ElixirLintingProvider.linterCommand, args, cmd.getOptions(vscode));
```

Comment:
Any thoughts about using `cwd` for the nearest parent directory containing `mix.exs` to get the actual command line output? Will that cover most use cases? Where will it not work?

Reply:
I think this would work well for most cases. The one place I can imagine it not working is if you have an umbrella project where credo is configured at the top level, but you pick up the mix file in a sub-application. I have a project like this, where credo is configured at the top level, so I can try it out

Reply:
I tried to find something in `mix` that would help like "Is this an umbrella project"? but I didn't find something like that.